### ISSUE TITLE
Breakpoints are off in the debugger

### ISSUE THREAD

**Issue by WenningQiu** [scriptcs/scriptcs#1103]
It appears that the breakpoints in Visual Studio might be off when a script has comments at the beginning of the file before using statements - the line of code that the debugger actually stops at does not appear to be what Visual Studio shows. When there is no using statements, or if the comments appear after the using statements, everything seems to be working fine.

The issue can be reprocuded using the script below:

//
// A simple demo script for scriptcs.
//

using System;
using System.Threading;

var content = "Hello, world!";

Console.WriteLine(content);

Thread.Sleep(5000);

**Comment by khellang** [comment]
Yeah, that's not what we'd consider a "valid" script. All usings, `#load` and `#r` should be before the script "body". The code in in question is this: https://github.com/scriptcs/scriptcs/blob/288e7d5c6b3f2e6d4bd91da54a60e1ce8cc07514/src/ScriptCs.Core/FilePreProcessor.cs#L118-L130

As you can see, we scan through each line until we hit a non-directive, non-parsable line of code. In your case, that's line no. 1. We could look into adding an extra check for comments.

**Comment by AmadeusW** [comment]
I fixed this issue by detecting comments `//` using a method similar to [`IsUsingLine`](https://github.com/scriptcs/scriptcs/blob/288e7d5c6b3f2e6d4bd91da54a60e1ce8cc07514/src/ScriptCs.Core/FilePreProcessor.cs#L150-L153), but multi line comments `/*` would require entirely different way of parsing the file. Not handling them would lead to inconsistency. What's the best thing to do?
